docs(en-US): protect your API on Spring Boot (#235)
logto-io/docs: docs/docs/recipes/protect-your-api/spring-boot.mdx

---
sidebar_label: Spring Boot
---

import TabItem from '@theme/TabItem';
import Tabs from '@theme/Tabs';

# Protect your API on Spring Boot

:::note

> This tutorial assumes you have created an API Resource `http://localhost:3000/protected` in Admin Console.

Reviewer: This could be misleading, `http://localhost:3000/` would be more common?

Author: good call! reorg the content. PTAL

---
sidebar_label: Spring Boot
---

import TabItem from '@theme/TabItem';
import Tabs from '@theme/Tabs';

# Protect your API on Spring Boot

:::note

This tutorial assumes you have created an API Resource `http://localhost:3000/` in Admin Console.
If you are not ready, [read this](./README.md#register-the-api-resources-through-logto-admin-console) before continuing.

:::

Your web application may run on the server-side using Spring Boot framework.
For now, you need to integrate Logto in Spring Boot manually.
This article guides you on how to finish it step by step.
And we use Gradle, Java, and Spring Security to take the example.

## Start a Spring Boot project

With [Spring Initializr](https://start.spring.io/), you can quickly start a Spring Boot project.

> Follow the options below:

Reviewer 2: nit
```suggestion
When prompt, make sure you use the following options:
```

public class ProtectedController {

    @GetMapping("/")
    public String protectedRoot() {
        return "Protected root.";
    }

    @GetMapping("/secret")
    public String protectedSecret() {
        return "Protected secret.";
    }

    @GetMapping("/profile")
    public String publicProfile() {
        return "Public profile.";
    }
}
```

## Access protected API

Build and run your Spring Boot web application, e.g. execute the `bootRun` Gradle task.

<Tabs>
<TabItem value="linux-or-macos" label="Linux or macOS">

```bash
./gradlew bootRun
```

</TabItem>
<TabItem value="windows" label="Windows">

```bash
gradlew.bat bootRun
```

</TabItem>
</Tabs>

:::note

This tutorial assumes you have the Access Token for an API resource `http://localhost:3000/` before making a request.
If you are not ready, [read this](./README.md#integrate-the-resources-authorization-flow-into-your-client-application) before continuing.

:::

Request your protected API with the Access Token as the Bearer token in the Authorization header,
e.g. execute the `curl` command.

```bash
curl --include 'http://localhost:3000/secret' \
--header 'Authorization: Bearer <your-access-token>'
```

> If succeeded, you will see the response like this.

Reviewer 2: nit:
```suggestion
If succeeded, you will get the response with 200 status.
```

```bash
HTTP/1.1 200
...
```

> Otherwise, it looks like that.

Reviewer 2: ```suggestion
Otherwise, the response would be something like this:
```
